src/date.cpp

```diff
@@ -27,7 +27,19 @@ Year      _cur_year;   ///< Current year, starting at 0
 Month     _cur_month;  ///< Current month (0..11)
 Date      _date;       ///< Current date in days (day counter)
 DateFract _date_fract; ///< Fractional part of the day.
-uint16 _tick_counter;  ///< Ever incrementing (and sometimes wrapping) tick counter for setting off various events
+uint16    _tick_counter;  ///< Ever incrementing (and sometimes wrapping) tick counter for setting off various events
+GameYear  _game_year;  ///< Number of years since the game began.
+
+bool _year_is_looping; ///< Is the current year repeating? (and has looped at least once)
+
+/**
+ * Checks if the current year is looping, and has occurred at least once.
```

Comment:
You can combine both statements into one to make it shorter:
"Returns whether the current year is selected to loop and has done so at least once."